scipy/linalg/tests/test_lapack.py

```diff
@@ -1645,3 +1645,115 @@ def test_getc2_gesc2():
+    np.random.seed(42)
+    for index, dtype in enumerate(DTYPES):
+        n = 10
+        if index % 2 == 0:
+            rtol = 10**-(np.finfo(dtype).precision-1)
+        else:
+            rtol = 10**-(np.finfo(dtype).precision-3)
+        if index < 2:
+            du = np.random.rand(n-1)
+            d = np.random.rand(n)
+            dl = np.random.rand(n-1)
+        else:
+            du = np.random.rand(n-1) + np.random.rand(n-1) * 1j
+            d = np.random.rand(n) + np.random.rand(n) * 1j
+            dl = np.random.rand(n-1) + np.random.rand(n-1) * 1j
+
+        diag_cpy = np.array([dl.copy(), d.copy(), du.copy()])
+
+        A = np.diag(d) + np.diag(dl, -1) + np.diag(du, 1)
+        x = np.random.rand(n)
+        b = A @ x
+
+        gttrf, gttrs = get_lapack_funcs(('gttrf', 'gttrs'), dtype=dtype)
+
+        _dl, _d, _du, du2, ipiv, info = gttrf(dl, d, du)
+
+        assert_array_equal(dl, diag_cpy[0])
+        assert_array_equal(d, diag_cpy[1])
+        assert_array_equal(du, diag_cpy[2])
+
+        perm_fnl = np.arange(n)
+        for i, piv in enumerate(ipiv):
+            perm_fnl[i], perm_fnl[piv-1] = perm_fnl[piv-1], perm_fnl[i]
+
+        _A = np.diag(_dl, -1) + np.diag(_d, 0) + np.diag(_du, 1)
+        #assert_allclose(_A, A[perm_fnl], rtol=rtol)
+
+        b_cpy = b.copy()
+        x_gttrs, info = gttrs(_dl, _d, _du, du2, ipiv, b)
+        assert_array_equal(b, b_cpy)
+
+        assert_allclose(x, x_gttrs, rtol=rtol)
+
+        if index < 2:
+            dl_misshaped = np.random.rand(n)
+            d_misshaped = np.random.rand(n-1)
+            du_misshaped = np.random.rand(n)
+        else:
+            dl_misshaped = np.random.rand(n) + np.random.rand(n) * 1j
+            d_misshaped = np.random.rand(n-1) + np.random.rand(n-1) * 1j
+            du_misshaped = np.random.rand(n) + np.random.rand(n) * 1j
+
+        with assert_raises(ValueError):
+            gttrf(dl_misshaped, d, du)
+        with assert_raises(ValueError):
+            gttrf(dl, d_misshaped, du)
+        with assert_raises(ValueError):
+            gttrf(dl, d, du_misshaped)
+
+        with assert_raises(Exception):
+            gttrf(dl[0], d[:1], du[0])
+
+        transpose_options = ["N", "C", "T"]
+        for trans in transpose_options:
+            if trans == "N":
+                b_trans = A @ x
+            elif trans == "T":
+                b_trans = np.transpose(A) @ x
+            elif trans == "C":
+                b_trans = A.conj().T @ x
+            _dl, _d, _du, du2, ipiv, info = gttrf(dl, d, du)
+            x_gttrs, info = gttrs(_dl, _d, _du, du2, ipiv, b_trans,
+                                  trans=trans)
+            assert_allclose(x, x_gttrs, rtol=rtol)
+
+        dl[0] = 0
+        du[0] = 0
+        d[0] = 0
+        __dl, __d, __du, _du2, _ipiv, _info = gttrf(dl, d, du)
+        np.testing.assert_(_info != 0,
+                           "?gttrf: singular matrix ret info == 0")
+
+    du = np.array([2.1, -1.0, 1.9, 8.0])
```

Comment:
Let's make this a separate test like `test_dttrf_NAG` so it only executes once (rather than once per data type). Also, please add a comment about the source of the test.

Reply:
Ok. My intent was to have it inside the test_gttrf_gttrs function, but outside the for loop. It might be indented one to many to be inside the for loop. Should I still make it a separate function?

Reply:
Oh, maybe your indentation is fine. I was thinking that the different datatypes were looped over by pytest, not an explicit loop in the function. 
Yes, I think it should be separated.
Thanks for dealing with the nitpicks. It's good to get this one perfect so we can just follow the example for the rest.

Reply:
Sounds good, I'm happy to deal with small things to make sure everything is exactly right. I separated it into a separate test function.